add training dropdown menu

milljm opened the pull request:
The content which will reside in this menu has not yet been finalized.

Closes #23569

Commit:
Add MOOSE Framework training

Adding a training menu which will be used to advertise upcoming
training.

This PR also moves menu items around to make room for a new "Training"
dropdown menu.

Closes #23569

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1408427/) on dfbe53a wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/23756/site)

This comment will be updated on new commits.

GiudGiud commented on modules/doc/config.yml:
```diff
@@ -40,29 +40,29 @@ Extensions:
         google-cse: 000781241768298771085:pedtnlxdkx8
         menu:
+            Training:
+                Comming Soon: /empty.md
```
```suggestion
                Coming Soon: /empty.md
```

GiudGiud commented on modules/doc/config.yml:
```diff
@@ -40,29 +40,29 @@ Extensions:
                 Application Development: application_development/index.md
                 Framework Development: framework_development/index.md
                 Modules: modules/index.md
-                NCRC Applications: help/inl/applications.md
```
this should stay?

milljm replied:
I suppose we can leave it. Its duplicated under Getting Started, so I figure meh.

milljm replied:
added it back.

GiudGiud commented on modules/doc/content/empty.md:
what is this

milljm replied:
I need to remove... It was a placeholder. No longer needed.

Commit:
Remove no longer needed file

Remove placeholder that was used for a no-content site the Training
menu was going to use.

milljm commented:
Not quite ready yet...
<img width="369" alt="Screenshot 2023-04-05 at 12 13 06 PM" src="https://user-images.githubusercontent.com/6414286/230168031-4d711921-71e5-4580-a165-0ab7598730b6.png">

Commit:
Shorten link title

For mobile devices, the link title was too long

milljm commented:
Documentation failed due to network related issues to civet... (pretty common). Please merge regardless.
